crates/storage-pg/migrations/20241004075132_queue_worker.sql

```diff
@@ -0,0 +1,37 @@
+-- Copyright 2024 New Vector Ltd.
+--
+-- SPDX-License-Identifier: AGPL-3.0-only
+-- Please see LICENSE in the repository root for full details.
+
+-- This table stores informations about worker, mostly to track their health
+CREATE TABLE queue_workers (
+  queue_worker_id UUID NOT NULL PRIMARY KEY,
+
+  -- When the worker was registered
+  registered_at TIMESTAMP WITH TIME ZONE NOT NULL,
+
+  -- When the worker was last seen
+  last_seen_at TIMESTAMP WITH TIME ZONE NOT NULL,
+
+  -- When the worker was shut down
+  shutdown_at TIMESTAMP WITH TIME ZONE
+);
+
+-- This single-row table stores the leader of the queue
+-- The leader is responsible for running maintenance tasks
+CREATE UNLOGGED TABLE queue_leader (
+  -- This makes the row unique
+  active BOOLEAN NOT NULL DEFAULT TRUE UNIQUE,
```

Comment:
I know it sounds silly, but I'd make this a `PRIMARY KEY` — maybe this sounds dogmatic? But there a handful of tools are not happy with tables that don't have a primary key, e.g. logical replication in Postgres by default, I'd say it's worth always using it instead of UNIQUE etc.

Reply:
Logical replication doesn't work anyway with UNLOGGED TABLE, but I'm happy to make that a PK, I don't think it really would change anything anyway

Reply:
ah true. But other tools may exist. I personally just treat it as good practice, so I'd still lightly vote in favour of PK instead of UNIQUE